oak/server/rust/oak_runtime/src/node/grpc/server/mod.rs

```diff
@@ -120,16 +131,26 @@ impl Node for GrpcServerNode {
         let channel_writer = GrpcServerNode::get_channel_writer(&runtime, handle)
             .expect("Couldn't initialize a channel writer");
 
-        let handler = HttpRequestHandler {
-            runtime,
-            writer: channel_writer,
+        // Handles incoming authentication gRCP requests
+        let auth_handler = match self.oidc_client_info {
+            Some(auth_config) => {
+                auth::oidc::build_service(&auth_config.client_id, &auth_config.client_secret)
+            }
+            // TODO(#1021): Add better handling to cases where the client info is not supplied.
+            _ => auth::oidc::build_service("", ""),
         };
 
         // Handles incoming TLS connections, unpacks HTTP/2 requests and forwards them to
         // [`HttpRequestHandler::handle`].
+        let http_handler = HttpRequestHandler {
+            runtime,
+            writer: channel_writer,
+        };
```

Comment:
The name of this seems a bit confusing, this is just the generic gRPC handler dispatcher, right?

Reply:
Renamed to `generic_handler`.